applications/CoSimulationApplication/python_scripts/data_transfer_operators/kratos_mapping.py

```diff
@@ -140,3 +143,208 @@ def __GetRankZeroModelPart():
+        else:
+            origin_settings["Parameters"]["nodal_solution_step_data_variables"].Append(variable_origin.Name())
+        if model_part_origin.NumberOfElements() == 0:
+            origin_settings["Parameters"]["entity_type"].SetString("condition")
+
+        # Create VTK output settings for the destination model part
+        destination_settings = self.__DefaultVTKSettings()
+        destination_settings["Parameters"]["model_part_name"].SetString(model_part_destination_name)
+        destination_settings["Parameters"]["output_path"].SetString("debug_vtk_output_" + variable_destination.Name() + "_" + identifier_destination)
+        if mapper_flags.Is(KM.Mapper.TO_NON_HISTORICAL):
+            destination_settings["Parameters"]["nodal_data_value_variables"].Append(variable_destination.Name())
+        else:
+            destination_settings["Parameters"]["nodal_solution_step_data_variables"].Append(variable_destination.Name())
+        if model_part_destination.NumberOfElements() == 0:
+            destination_settings["Parameters"]["entity_type"].SetString("condition")
+
+        return model_origin, model_destination, origin_settings, destination_settings, identifier_tuple
+
+    def __GenerateProcessVTK(self, from_solver_data, to_solver_data, transfer_options):
+        """
+        Execute VTK output processing using provided settings on the specified model part.
+
+        Args:
+            self: The instance of the class.
+            from_solver_data (CoSimulationData): The data from the solver to transfer from.
+            to_solver_data (CoSimulationData): The data from the solver to transfer to.
+            transfer_options (KM.Flags): The flags to be used for the mapping.
+        """
+        if self.debug_vtk:
+            model_origin, model_destination, origin_settings, destination_settings, identifier_tuple = self.__PrepareSettings(from_solver_data, to_solver_data, transfer_options)
+
+            # Get the output paths
+            origin_output_path = origin_settings["Parameters"]["output_path"].GetString()
+            destination_output_path = destination_settings["Parameters"]["output_path"].GetString()
+
+            # Copy settings for pre and post mapping
+            pre_origin_settings = KM.Parameters(origin_settings.PrettyPrintJsonString())
+            post_origin_settings = KM.Parameters(origin_settings.PrettyPrintJsonString())
+            pre_destination_settings = KM.Parameters(destination_settings.PrettyPrintJsonString())
+            post_destination_settings = KM.Parameters(destination_settings.PrettyPrintJsonString())
+
+            # Set the output paths
+            pre_origin_settings["Parameters"]["output_path"].SetString(origin_output_path + "_pre_map")
+
+            # Create a VTK output process object with the provided settings and the current model
+            pre_process_origin = vtk_output_process.Factory(pre_origin_settings, model_origin)
```

Comment:
I would use the vtk_output directly, then you only need to call `PrintOutput`, and not have all the overhead of the process

Reply:
I understand your point, but the problem with that impliementation is that I will need to manually add the settings for the IO instead of automatically deduce the variables and settings from the mapping like I do in the current proposal.

Reply:
Ping @philbucher